### PR TITLE
Fix zellic finding 5

### PR DESCRIPTION
**Pull request by kunxian-xia**
### Description

Fix the finding 5 reported in [Zellic/Kalos Audit Report finding 5](https://hackmd.io/4XuxkwHERa6AIPQ6tr2QAg#Finding-5-RLP-Circuit-%E2%80%9CImportance%E2%80%9D-Medium).

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

File: zkevm-circuits/src/rlp_circuit_fsm.rs [hunk 1]
```diff
@@ -1121,6 +1148,14 @@ impl<F: Field> RlpCircuitConfig<F> {
             // Bytes => DecodeTagStart
             cb.condition(tidx_eq_tlen, |cb| {
                 // assertions
+                let (lt, eq) = tlength_lte_mlength.expr(meta, Some(Rotation::cur()));
+                cb.require_equal(
+                    "tag_length <= max_length",
+                    sum::expr([
```
Comment (roynalnaruto) [inline zkevm-circuits/src/rlp_circuit_fsm.rs:1158]: nitpick: I would personally prefer `or` over `sum` as its more intuitive in this case

File: zkevm-circuits/src/witness/rlp_fsm.rs [hunk 2]
```diff
@@ -151,8 +151,8 @@ use crate::{
     },
 };
 
-// The number of bytes of list can not larger than 2^24.
-pub(crate) const N_BYTES_LIST: usize = 1 << 24;
+// The number of bytes of list can not larger than 2^24 = 2^(8*3).
+pub(crate) const N_BYTES_LIST: usize = 3;
```
Comment (roynalnaruto) [inline zkevm-circuits/src/witness/rlp_fsm.rs]: why this change?

File: zkevm-circuits/src/rlp_circuit_fsm.rs (hunk 1 above)
Comment (kunxian-xia) [reply zkevm-circuits/src/rlp_circuit_fsm.rs:1158]: But `or` has higher degree than `sum`. How about I leave some comments there?

File: zkevm-circuits/src/witness/rlp_fsm.rs (hunk 2 above)
Comment (kunxian-xia) [reply zkevm-circuits/src/witness/rlp_fsm.rs]: Because if we want to add `tag_length <= max_length` for the case of `LongList`, then the `tag_length` here means the number of bytes to represent a `LongList`(e.g. `[0xf9, 0x01, 0xff]` has `tag_length = 2`). Maybe we should change the naming of this const?

Comment (roynalnaruto) [reply zkevm-circuits/src/witness/rlp_fsm.rs]: Yes, and the comment as well can be more descriptive..

File: zkevm-circuits/src/rlp_circuit_fsm.rs (hunk 1 above)
Comment (roynalnaruto) [reply zkevm-circuits/src/rlp_circuit_fsm.rs:1158]: Okay, it works in this case as both `lt` and `eq` are boolean, so `sum` and `or` will actually give us the same result. But yes, a comment to explain it would be better as `sum` by itself shouldn't be used (but it works in this case).

**Review by roynalnaruto — APPROVED** [review]
